incubator/bitcoincashd/templates/deployment.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: {{ template "bitcoincashd.fullname" . }}
+  labels:
+    app: {{ template "bitcoincashd.name" . }}
+    chart: {{ template "bitcoincashd.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "bitcoincashd.name" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "bitcoincashd.name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      {{- if .Values.configurationFile }}
+      initContainers:
+        - name: copy-bitcoincashd-config
+          image: busybox
+          command: ['sh', '-c', 'cp /configmap/bitcoincash.conf /root/.bitcoin/bitcoincash.conf']
+          volumeMounts:
+            - name: configmap
+              mountPath: /configmap
+            - name: config
+              mountPath: /root/.bitcoin/
+      {{- end }}
+      containers:
+        - name: {{ template "bitcoincashd.fullname" . }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          resources:
+  {{ toYaml .Values.resources | indent 10 }}
+          ports:
+            - name: rpc
+              containerPort: {{ .Values.service.rpcPort }}
```

Comment:
It probably doesn't make sense to have these configurable like this. Changing the service ports also changes container ports which is likely to break things. I'd suggest you hard-code container ports.